lib/assertions.js

```diff
@@ -786,7 +786,15 @@ module.exports = (expect) => {
             );
           });
           if (expect.flags.only) {
-            expect(subject.length, 'to be', args.length);
+            expect(
+              subject &&
+                Array.prototype.every.call(subject, (item) =>
+                  Array.prototype.some.call(args, (arg) =>
+                    expect.equal(item, arg)
+                  )
+                ),
+                'to be truthy'
+            );
```

Comment:
Seems reasonable 👍

Reply:
Sorry, I started my own review thread at the same place. I will move my comments here :-)

--- 

> @sunesimonsen:
> You need to run through the RHS items, for each of them you check if it matches each of the items in the LHS. You record that information in an array of boolean, where you always or the results. If the RHS item doesn't match any items you fail. When you are done you check that every item in the array is true; otherwise you fail.
>
> I think the above would work, but I also realise that I can't ask you to go an do that, so I'll probably volunteer to do it. But I guess you need to solution now, so that is a bit of pickle 😅

If I understand your description correctly, I did the almost the opposite here. I run through all the items in the LHS, making sure that each of them equals at least one of the specs on the RHS. That was the simplest way I could come up with on the spot.

Does it have significantly worse performance characteristics than your proposed solution? If so I'm all for taking a stab at inversing it :-)

-- 

Of I think I just realized something! You wanted it to fail if one of the items on the RHS is not used in the subject. Let me express it with a test:

```js
expect([1,1,1], 'to only contain', 1, 2)
```

That would pass my implementation.

Does the array `[1, 1, 1]` contain only the elements `1` and `2`? I guess it sort of does, but it also doesn't...

Reply:
I think we shouldn't worry too much about the performance, we can always optimise it if it performs poorly.

Reply:
> Does the array [1, 1, 1] contain only the elements 1 and 2?

Hehe 😅 - yes I guess you could read it that way. Maybe that is the expected semantic. What do you say @papandreou?

I think there is still a use case for an assertion then, that says this array should the following items in any order. 

Is that just `expect(new Set([0,3,1,2]), 'to satisfy', new Set([0,1,2,3]))`?

Reply:
> @papandreou 27 minutes ago
> My take: It doesn't only contain the elements 1 and 2 because it doesn't contain 2. To me, the only bit is an extra requirement.

Reply:
@papandreou I think I would agree completely with you, if it was still `to [exhaustively] contain`. I read `to only contain` similarly to `to be composed of the following elements`. The "only" qualifier means that items that do not look like the RHS may not be in the LHS - but I am not sure that I think it implies that all items on the RHS must be represented in the LHS?

Reply:
`to [only] [exhaustively] contain` ... 🙈  🙉  🙊

Reply:
I think it would be super surprising if that wasn't implied. Then `expect([], 'to only contain', 1)` would also pass.

Reply:
I am convinced.

Btw, I also misspoke earlier - my implementation for the only-flag would indeed let `expect([], 'to only contain', 1)` pass, but the base check for contain will fail if items on the RHS is not in the LHS.

```js
expect([1, 1, 1, 1], 'to only contain', 1, 2);
```

```
expected [ 1, 1, 1, 1 ] to only contain 1, 2

[
  1,
  1,
  1,
  1
  // missing 2
]
```

The error is produced by the first check in the `to contain` implementation, but the diff is generated by the only-flag branch. If you do the same assertion without the only flag you don't get a diff:

```js
expect([1, 1, 1, 1], 'to contain', 1, 2);
```

```
expected [ 1, 1, 1, 1 ] to contain 1, 2
```

Reply:
Given that this is already established behavior in `to contain` I think it's a dangerous idea to change semantics in `to only contain`. I will resolve this conversation, as I think we covered all the bases.

Reply:
@gustavnikolaj I agree with you about that.